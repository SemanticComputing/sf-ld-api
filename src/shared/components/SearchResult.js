import React from 'react';
import _ from 'lodash';
import $ from 'jquery';
import ReactTooltip from 'react-tooltip';
import eli from '../lib/eli';
import { string } from 'prop-types';

export default class SearchResult extends React.Component {

  constructor(props) {
    const showContentTypes = [
      'http://data.finlex.fi/schema/sfl/Section',
      'http://data.finlex.fi/schema/sfl/Subsection'
    ];
    super(props);
    this.sectionId = eli.getSectionOfALawLocalId(props.workUrl);
    this.statuteId = eli.getStatuteLocalId(props.workUrl);
    this.statuteTitle = props.statuteTitle || '';
    this.heading = props.title;
    this.content = (_.indexOf(showContentTypes, props.type) > -1) ? this.sanitize(props.content, props.query) : '';
    this.versionUrl = props.versionUrl;
    this.workUrl = props.workUrl;
    this.statuteVersionUrl = props.statuteVersionUrl;
  }

  sanitize(html, query) {
    const $html = $('<div />', {html:html});
    $html.find('.item-identifier').html('');
    $html.find('.reference-amendment').html('');
    $html.find('.heading').html('');
    $html.find('.section-id').html('');
    $html.html($html.html().replace(new RegExp(query,'gi'), '<strong>$&</strong>'));
    return $html.html();
  }

  render() {
    return (
      <div className="search-result">
        <div className="search-result-statute-id"><a href={this.statuteVersionUrl ? this.statuteVersionUrl : this.versionUrl}>{this.statuteId}</a></div>
        <div className="search-result-statute-title"><a href={this.statuteVersionUrl ? this.statuteVersionUrl : this.versionUrl}>{this.statuteTitle}</a></div>
        <div className="search-result-title"><a href={this.versionUrl}>{this.sectionId} {this.heading && this.sectionId ? '- '+this.heading : this.heading ? this.heading : ''}</a></div>
        <div className="search-result-text" dangerouslySetInnerHTML={{__html: this.content}}></div>
        <ReactTooltip />
        <hr/>
      </div>
    );
  }
}

SearchResult.propTypes = {
  workUrl: string,
  content: string,
  versionUrl: string,
  statuteVersionUrl: string,
  type: string,
  query: string,
  title: string,
  statuteTitle: string
};
