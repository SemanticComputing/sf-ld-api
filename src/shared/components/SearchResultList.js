import React from 'react';
import { map } from 'lodash';
import SearchResult from './SearchResult';
import { string, array } from 'prop-types';

export default function SearchResulList(props) {

  function renderResults(results, query) {
    return map(results, (result, idx) => {
      return <SearchResult
        key={idx + '-' + new Date().getTime()}
        title={result.title ? result.title.value : ''}
        content={result.txt ? result.txt.value : ''}
        query={query}
        workUrl={result.s ? result.s.value : ''}
        versionUrl={result.v ? result.v.value : ''}
        statuteVersionUrl={result.st ? result.st.value : ''}
        statuteTitle={result.stt ? result.stt.value : ''}
        type={result.t ? result.t.value : ''}
      >
      </SearchResult>;
    });
  }

  return (
    <div className="search-results">
      {renderResults(props.results, props.query)}
    </div>
  );
}

SearchResulList.propTypes = {
  results: array,
  query: string
};
