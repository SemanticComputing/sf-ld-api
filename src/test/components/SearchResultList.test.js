import React from 'react';
import { shallow } from 'enzyme';

import SearchResultList from '../../shared/components/SearchResultList';
import SearchResult from '../../shared/components/SearchResult';
import { rikosSearchResult } from '../fixtures';

describe('<SearchResultList />', () => {
  it('renders <SearchResult /> components based on the given results', () => {
    const wrapper = shallow(<SearchResultList results={rikosSearchResult} query="rikos" />);
    expect(wrapper.find(SearchResult).length).toBe(3);
    expect(wrapper).toMatchSnapshot();
  });
});
