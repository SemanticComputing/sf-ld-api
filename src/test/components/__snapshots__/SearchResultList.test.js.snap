// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultList /> renders <SearchResult /> components based on the given results 1`] = `
<div
  className="search-results"
>
  <SearchResult
    content="<div class=\\"section\\"><h4 class=\\"item-identifier\\">7 §</h4><span class=\\"reference reference-amendment\\"><a href=\\"http://data.finlex.fi/eli/sd/1996/626/alkup\\">(16.8.1996/626)</a></span>
  <h4 class=\\"heading\\">Kansainvälinen rikos</h4>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Suomen ulkopuolella tehtyyn rikokseen, jonka rankaiseminen tekopaikan laista riippumatta perustuu Suomea velvoittavaan kansainväliseen sopimukseen tai muuhun Suomea kansainvälisesti velvoittavaan säädökseen tai määräykseen <i>(kansainvälinen rikos)</i>, sovelletaan Suomen lakia. Tarkemmat säännökset tämän pykälän soveltamisesta annetaan asetuksella.</p>
  </div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Suomen lakia sovelletaan tekopaikan laista riippumatta myös sellaiseen Suomen ulkopuolella tehtyyn ydinräjähderikokseen tai yleisvaarallisen rikoksen valmisteluun, jota on pidettävä täydellisestä ydinkoekiellosta tehdyssä sopimuksessa (SopS 15/2001) tarkoitettuna rikoksena.</p>
  <span class=\\"reference reference-amendment\\"><a href=\\"http://data.finlex.fi/eli/sd/1998/841/alkup\\">(30.12.1998/841 v. 2003)</a></span>
  </div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Suomen lakia sovelletaan tekopaikan laista riippumatta myös Suomen ulkopuolella tehtyyn ihmiskauppaan, törkeään ihmiskauppaan ja 34 a luvussa tarkoitettuun rikokseen.</p>
  <span class=\\"reference reference-amendment\\"><a href=\\"http://data.finlex.fi/eli/sd/2004/650/alkup\\">(9.7.2004/650)</a></span>
  </div>
  </div>"
    query="rikos"
    statuteTitle="Rikoslaki"
    statuteVersionUrl="http://data.finlex.fi/eli/sd/1889/39/ajantasa/20160101"
    title="Kansainvälinen rikos"
    type="http://data.finlex.fi/schema/sfl/Section"
    versionUrl="http://data.finlex.fi/eli/sd/1889/39/luku/1/pykala/7/ajantasa/19960901"
    workUrl="http://data.finlex.fi/eli/sd/1889/39/luku/1/pykala/7"
  />
  <SearchResult
    content="<div class=\\"section\\"><h4 class=\\"item-identifier\\">27 §</h4>
  <h4 class=\\"heading\\">Ammattitoiminnassa tehty rikos</h4>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Jos terveydenhuollon ammattihenkilö on tuomittu vankeusrangaistukseen rikoksesta, jonka hän on tehnyt ammattitoiminnassaan, ja tuomioistuimen päätös on saanut lainvoiman ja jos rikokseen liittyvistä asianhaaroista on havaittavissa, ettei hän ole sen luottamuksen arvoinen, jota hänen tulee nauttia, Sosiaali- ja terveysalan lupa- ja valvontavirasto voi laillistetulta ammattihenkilöltä poistaa määräajaksi tai, jos asianhaarat ovat erittäin raskauttavat, lopullisesti ammatinharjoittamisoikeuden tai luvan saaneelta ammattihenkilöltä peruuttaa ammatinharjoittamisluvan määräajaksi tai, jos asianhaarat ovat erittäin raskauttavat, lopullisesti taikka kieltää nimikesuojattua ammattihenkilöä määräajaksi tai jos asianhaarat ovat erittäin raskauttavat, lopullisesti käyttämästä asetuksella säädettyä terveydenhuollon ammattihenkilön ammattinimikettä.</p>
  <span class=\\"reference reference-amendment\\"><a href=\\"http://data.finlex.fi/eli/sd/2015/262/alkup\\">(20.3.2015/262)</a></span>
  </div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Jos valtion tai kunnan taikka kuntayhtymän virassa oleva terveydenhuollon ammattihenkilö on rikoksen johdosta tuomittu erotettavaksi virantoimituksesta tai viralta pantavaksi, noudatetaan vastaavasti, mitä 1 momentissa säädetään.</p>
  </div>
  <div class=\\"subsection\\">
  <p class=\\"text-element\\">3 momentti on kumottu L:lla <span class=\\"reference reference-repeal\\"><a href=\\"http://data.finlex.fi/eli/sd/2015/262/alkup\\">20.3.2015/262</a></span>.</p></div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Jo ennen kuin tuomioistuimen päätös, jolla terveydenhuollon ammattihenkilö on tuomittu vankeusrangaistukseen taikka viralta pantavaksi tai virantoimituksesta erotettavaksi, on saanut lainvoiman, Sosiaali- ja terveysalan lupa- ja valvontavirasto voi kieltää laillistettua ammattihenkilöä harjoittamasta ammattia tai peruuttaa luvan saanen ammattihenkilön ammatinharjoittamisluvan taikka kieltää nimikesuojattua ammattihenkilöä käyttämästä asetuksella säädettyä terveydenhuollon ammattihenkilön ammattinimikettä.</p>
  <span class=\\"reference reference-amendment\\"><a href=\\"http://data.finlex.fi/eli/sd/2015/262/alkup\\">(20.3.2015/262)</a></span>
  </div>
  </div>"
    query="rikos"
    statuteTitle="Laki terveydenhuollon ammattihenkilöistä"
    statuteVersionUrl="http://data.finlex.fi/eli/sd/1994/559/ajantasa/20160101"
    title="Ammattitoiminnassa tehty rikos"
    type="http://data.finlex.fi/schema/sfl/Section"
    versionUrl="http://data.finlex.fi/eli/sd/1994/559/luku/5/pykala/27/ajantasa/20150401"
    workUrl="http://data.finlex.fi/eli/sd/1994/559/luku/5/pykala/27"
  />
  <SearchResult
    content="<div class=\\"section\\"><h4 class=\\"item-identifier\\">25 §</h4><span class=\\"reference reference-amendment\\"><a href=\\"http://data.finlex.fi/eli/sd/2006/301/alkup\\">(28.4.2006/301)</a></span>
  <h4 class=\\"heading\\">Ammattitoiminnassa tehty rikos</h4>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Jos eläinlääkärinammatin harjoittaja on tuomittu vankeusrangaistukseen rikoksesta, jonka hän on tehnyt ammattitoiminnassaan, ja tuomioistuimen tuomio on saanut lainvoiman ja jos rikokseen liittyvistä asianhaaroista on havaittavissa, ettei hän ole sen luottamuksen arvoinen, jota hänen tulee nauttia, Elintarviketurvallisuusvirasto voi poistaa eläinlääkärinammatin harjoittajalta määräajaksi tai, jos asianhaarat ovat erittäin raskauttavat, lopullisesti ammatinharjoittamisoikeuden.</p>
  </div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Jos valtion tai kunnan taikka kuntayhtymän virassa oleva eläinlääkärinammatin harjoittaja on rikoksen johdosta tuomittu erotettavaksi virantoimituksesta tai viralta pantavaksi, noudatetaan vastaavasti, mitä 1 momentissa säädetään.</p>
  </div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Tuomioistuimen tulee viipymättä lähettää jäljennös 1 ja 2 momentissa tarkoitettua asiaa koskevasta pöytäkirjasta ja tuomiosta Elintarviketurvallisuusvirastolle.</p>
  </div>
  <div class=\\"subsection\\"><p class=\\"text-element\\">Jo ennen kuin tuomioistuimen tuomio, jolla eläinlääkärinammatin harjoittaja on tuomittu vankeusrangaistukseen taikka viralta pantavaksi tai virantoimituksesta erotettavaksi, on saanut lainvoiman, Elintarviketurvallisuusvirasto voi kieltää eläinlääkärinammatin harjoittajaa harjoittamasta ammattia.</p>
  </div>
  </div>"
    query="rikos"
    statuteTitle="Laki eläinlääkärinammatin harjoittamisesta"
    statuteVersionUrl="http://data.finlex.fi/eli/sd/2000/29/ajantasa/20160101"
    title="Ammattitoiminnassa tehty rikos"
    type="http://data.finlex.fi/schema/sfl/Section"
    versionUrl="http://data.finlex.fi/eli/sd/2000/29/luku/5/pykala/25/ajantasa/20060501"
    workUrl="http://data.finlex.fi/eli/sd/2000/29/luku/5/pykala/25"
  />
</div>
`;
